### PR TITLE
Login Error Handling + Tests

### PR DESCRIPTION
**Pull request by csc207jc**
This small pull request adds error handling to login.
It also adds a few tests to the login page.

### CODE REVIEW

File: tests/LoginTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import interface_adapter.ViewManagerModel;
+import interface_adapter.login.LoginViewModel;
+import javax.swing.*;
+import okhttp3.OkHttpClient;
+import org.junit.Assert;
+import org.junit.Test;
+import view.LoginView;
+import view.WelcomeView;
+
+public class LoginTest extends ButtonTest {
```
Comment (uoftm) [inline tests/LoginTest.java:13]: Yay, tests!

File: src/data_access/FirebaseUserDataAccessObject.java [hunk 2]
```diff
@@ -61,45 +50,33 @@ public User get(String email, String password) {
...
-        System.out.println("Authentication failed: " + authResponseData);
-        return null;
+        String failureMessage = authResponseJson.getJSONObject("error").getString("message");
+        throw new RuntimeException("Authentication failed: " + failureMessage);
       }
 
-      JSONObject authResponseJson = new JSONObject(authResponseData);
+      if (!authResponse.isSuccessful()) {
+        String failureMessage = authResponseJson.getJSONObject("error").getString("message");
+        throw new RuntimeException("Authentication failed: " + failureMessage);
```
Comment (alexrosen45) [inline src/data_access/FirebaseUserDataAccessObject.java:62]: We should map common Firebase error messages to something nicer to display for the user.

File: tests/WelcomeTest.java [hunk 3]
```diff
@@ -1,68 +1,32 @@
...
 import org.junit.Test;
-import use_case.switch_view.SwitchViewInputBoundary;
-import use_case.switch_view.SwitchViewInteractor;
-import use_case.switch_view.SwitchViewOutputBoundary;
 import view.LoginView;
 import view.SignupView;
 import view.WelcomeView;
 
-public class WelcomeTest {
+public class WelcomeTest extends ButtonTest {
```
Comment (alexrosen45) [inline tests/WelcomeTest.java:10]: nice

**Review by alexrosen45 — APPROVED** [review]
We can change small stuff like the exact wording of error messages later -- approved!

**Review by alexrosen45 — APPROVED** [review]

**Pull request merged**